docusaurus/docs/introduction/plugin-lifecycle.md

```diff
@@ -0,0 +1,93 @@
+---
+id: plugin-lifecycle
+title: Life cycle of a plugin
+description: Learn about the life cycle of a Grafana plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - lifecycle
+  - life cycle
+  - loading
+  - unloading
+  - installation
+sidebar_position: 2.5
+---
+
+# Life cycle of a plugin
+
+This document describes the various phases of a plugin including installation and loading. We will describe the differences in a plugin's life cycle depending on its type and whether or not it has a backend.  
+
+:::note
+
+The life cycle of plugins is tracked in-memory and is not persisted in Grafana’s database. 
+
+:::
+
+## Installing and uninstalling a plugin
+
+For instructions to install or uninstall plugins, see our documentation at [Plugin administration](https://grafana.com/docs/grafana/latest/administration/plugin-management/#install-grafana-plugins).
+
+Upon installation, a plugin is extracted to the _plugin directory_ on the filesystem. Similarly, uninstalling a plugin deletes the files from the same directory.  
+
+:::note
+
+Use of the Grafana CLI to install or uninstall a plugin requires you to restart Grafana for the change to take effect. To avoid restarting Grafana, you can install a plugin from within the Grafana [plugin catalog](https://grafana.com/plugins/) during runtime.
+
+:::
+
+## Loading plugins
+
+Understanding the different phases involved when Grafana is starting a plugin may help you better understand plugin usage and to debug any unexpected behavior during loading.
+
+### Phase 1. Plugin initialization
+
+All plugins are initialized either when Grafana starts up or when a plugin has been installed/uninstalled during runtime.
+
+For [backend](./backend.md) plugins, there is an additional initialization process (see [Phase 4](#phase-4-backend-plugin-initialization)).
+
+### Phase 2. Plugin discovery 
+
+Grafana starts to discover which plugins are installed by scanning directories on the file system for every `plugin.json`.
+
+### Phase 3. Plugin loading 
+
+All plugins that were discovered in the discovery phase are checked to make sure they’re valid and that they have a valid signature. Valid plugins are referred to as _verified plugins_. 
```

Comment:
```suggestion
All plugins that were discovered in the discovery phase are checked to make sure they’re valid and that they have a valid [signature](https://grafana.com/docs/grafana/latest/administration/plugin-management/#plugin-signatures). Valid plugins are referred to as _verified plugins_. 
```

Reply:
Beyond a signature check, what else is validated?

Reply:
Not sure we need to mention that in detail. But to mention what we currently have
- Logs a warning message if a module.js file is missing
- Angular detection: If angular support is disabled and angular detected in plugin we record an error and doesn't allow this plugin to be loaded

Reply:
I didnn't add module.js error, but I added Angular deprecation because I think that's something that we want to publicize whenever it makes sense.